src/main/java/com/adobe/target/edge/client/service/TelemetryService.java

```diff
@@ -46,12 +47,25 @@ public void addTelemetry(
       TargetDeliveryRequest deliveryRequest,
       TimingTool timer,
       TargetDeliveryResponse targetDeliveryResponse) {
+    if (!DefaultRuleLoader.artifactDownloadTimeStore.isEmpty()
```

Comment:
I think we should use a separate overloaded `addTelemetry` method specifically for the artifact download entry.  Because it really isn't related to the Delivery request entry at all.  I think we should also pass the TelemetryService instance into the RuleLoader instance and directly call telemetryService.addTelemetry() from there.  This way TelemetryService can stay dumb and won't need to know about RuleLoader.  And it also removes the need for the artifactDownloadTimeStore inside the RuleLoader.